src/supplementField.ts

```diff
@@ -243,7 +243,7 @@ export function supplementFieldOption(fieldOption: FieldOption, fieldDef: FieldD
   }
 
   // supplement bin from fieldDef, user should never have to provide bin in options
-  if (fieldDef.bin) {
+  if (fieldDef.bin && typeof supplementedFieldOption.field !== 'function') {
```

Comment:
I think it might be more accurate if we use `isString(supplementedFieldOption.field)` to check the condition directly.